File: app_java/src/main/java/gorden/jrefresh/demo/MainActivity.java
package gorden.jrefresh.demo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import gorden.refresh.JRefreshLayout;

public class MainActivity extends AppCompatActivity {
    JRefreshLayout refreshLayout;
    RecyclerView recyclerView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        refreshLayout = (JRefreshLayout) findViewById(R.id.refreshLayout);
        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);

        recyclerView.setAdapter(new RecyclerView.Adapter() {
            @Override
            public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
                return new RecyclerView.ViewHolder(LayoutInflater.from(getBaseContext()).inflate(R.layout.item_sample,null)) {
                };
            }

            @Override
            public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

            }

            @Override
            public int getItemCount() {
                return 20;
            }
        });

        refreshLayout.setJRefreshListener(new JRefreshLayout.JRefreshListener() {
            @Override
            public void onRefresh(final JRefreshLayout refreshLayout) {
                refreshLayout.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        refreshLayout.refreshComplete(true);
                    }
                },3000);
            }
        });

    }
}

File: app/src/main/kotlin/gorden/krefreshlayout/demo/ui/MainActivity.kt
package gorden.krefreshlayout.demo.ui

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.SimpleAdapter
import gorden.krefreshlayout.demo.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    val gridItems = arrayOf("RecyclerView", "ScrollView", "NestedScrollView", "ViewPager in NestedScrollView", "ViewPager in ScrollView"
            , "WebView ", "Fragment in ViewPager", "CoordinatorLayout", "JAVA版本", "gordenxqw@gmail.com", "QQ:354419188")
    val dataList = ArrayList<Map<String, String>>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        for (item in gridItems) {
            val map = HashMap<String, String>()
            map.put("name", item)
            dataList.add(map)
        }
        gridView.adapter = SimpleAdapter(this, dataList, R.layout.item_main, arrayOf("name"), intArrayOf(R.id.textView))

        gridView.setOnItemClickListener { parent, view, position, id ->
            val intent = Intent("SampleActivity")
            intent.putExtra("position", position);
            intent.putExtra("name", gridItems[position])
            startActivity(intent)
        }

        refreshLayout.setKRefreshListener {
            refreshLayout.postDelayed({
                refreshLayout.refreshComplete(true)
            }, 2000)
        }

//        refreshLayout.startRefresh()
    }
}
